django/template/exceptions.py

```diff
@@ -0,0 +1,31 @@
+class TemplateDoesNotExist(Exception):
+    """
+    The central exception used by backends when a template does not exist.
+    Accepts the following optional arguments:
+
+    backend
+        The template backend class used when raising this exception.
+
+    tried
+        A list of sources that were tried when finding the template. This
+        is formatted as a list of tuples containing (origin, status), where
+        origin is an Origin object and status is a string with the reason the
+        template wasn't found.
+
+    chain
+        A list of intermediate TemplateDoesNotExist exceptions. This is used to
+        encapsulate multiple exceptions when loading templates from multiple
+        engines.
+    """
+    def __init__(self, msg, tried=None, backend=None, chain=None):
+        self.backend = backend
+        self.tried = tried or []
```

Comment:
Can you check explicitly?

    if tried is None:
        tried = []
    self.tried = tried

This is a good practice in general to avoid hard-to-diagnose errors if an unexpected, falsy value of `tried` is passed in.